File: [프로그래머스] SummerWinter Coding2018_스킬트리.py
def solution(skill, skill_trees):
    answer = 0
    for i in skill_trees:
        s_list = list(skill)
        for j in i:
            if j in skill:
                if j != s_list.pop(0):
                    break
        else:
            answer += 1
    return answer

File: [프로그래머스] SummerWinter Coding2018_스킬트리.js
function solution(skill, skill_trees) {
    var answer = 0;
    for(let i of skill_trees){
        let arr = skill.split('');
        let flag = 1;
        for(let j of i){
            if(arr.includes(j)){
                if(j !== arr.shift()){
                    flag = 0;
                    break;
                }
            }
            
        }
        if(flag === 1)  {answer++;}
    }
    return answer;
}